packages/services/src/kernel/default.ts

```diff
@@ -251,16 +270,32 @@ export class KernelConnection implements Kernel.IKernelConnection {
     if (this.isDisposed) {
       return;
     }
-    this._isDisposed = true;
-    this._disposed.emit();
 
-    this._updateConnectionStatus('disconnected');
-    this._clearKernelState();
-    this._pendingMessages = [];
-    this._clearSocket();
+    const cleanup = () => {
+      this._isDisposed = true;
+      this._disposed.emit();
 
-    // Clear Lumino signals
-    Signal.clearData(this);
+      this._updateConnectionStatus('disconnected');
+      this._clearKernelState();
+      this._pendingMessages = [];
+      this._clearSocket();
+
+      // Clear Lumino signals
+      Signal.clearData(this);
+    };
+
+    if (this._subshellId !== null) {
+      const future = this.requestDeleteSubshell(
+        { subshell_id: this._subshellId },
```

Comment:
You will need to add `subshell_id` to this list:

https://github.com/jupyterlab/jupyterlab/blob/b30baa1f275de1515bee4d1926866c1bb292b32e/.eslintrc.js#L76-L83

Reply:
Thanks.